lib/compat/wordpress-6.0/block-patterns.php

```diff
@@ -98,31 +98,38 @@ function gutenberg_register_theme_block_patterns() {
 
 	foreach ( $themes as $theme ) {
 		$dirpath = $theme->get_stylesheet_directory() . '/patterns/';
+		if ( ! is_dir( $dirpath ) || ! is_readable( $dirpath ) ) {
+			continue;
+		}
 		if ( file_exists( $dirpath ) ) {
```

Comment:
nit: is this `file_exists` check still necessary after the "`$dirpath` is a readable directory" check has passed?